auth/realm/base/src/main/java/org/wildfly/security/auth/realm/AggregateSecurityRealm.java

```diff
@@ -88,12 +101,18 @@ public RealmIdentity getRealmIdentity(final Evidence evidence) throws RealmUnava
     public RealmIdentity getRealmIdentity(final Principal principal) throws RealmUnavailableException {
         boolean ok = false;
         final RealmIdentity authenticationIdentity = authenticationRealm.getRealmIdentity(principal);
+
+        Principal authorizationPrincipal = principal;
+        if (principalTransformer != null) {
+            authorizationPrincipal = principalTransformer.apply(authorizationPrincipal);
+            if (authorizationPrincipal == null) throw new IllegalStateException();
```

Comment:
Should be using ```throw ElytronMessages.log.someMethodName()``` here to also provide a message that indicates that the authorization principal was null after the transformation. Take a look at ```org.wildfly.security.auth.realm.ElytronMessages``` to see examples on how to do this.

Reply:
Fixed